crates/noirc_evaluator/src/ssa_refactor/acir_gen/mod.rs

```diff
@@ -238,7 +241,9 @@ impl Context {
             }
             self.ssa_values.insert(*param_id, value);
         }
-        Ok(())
+        // A range does not include the last element
+        let end_witness = self.acir_context.current_witness_index().0 + 1;
+        Ok(start_witness..end_witness)
```

Comment:
You could change this to an inclusive range (`RangeInclusive`)
```suggestion
        let end_witness = self.acir_context.current_witness_index().0;
        Ok(start_witness ..= end_witness)
```

Reply:
Done